src/sql/workbench/api/node/extHostQueryEditor.ts

```diff
@@ -57,8 +57,12 @@ export class ExtHostQueryEditor implements ExtHostQueryEditorShape {
 	public $onQueryEvent(handle: number, fileUri: string, event: IQueryEvent): void {
 		let listener: azdata.queryeditor.QueryEventListener = this._queryListeners[handle];
 		if (listener) {
-			let planXml = event.params ? event.params.planXml : undefined;
-			listener.onQueryEvent(event.type, new ExtHostQueryDocument(mssqlProviderName, fileUri, this._proxy), planXml);
+			if(event.type === 'visualize') {
```

Comment:
you could probably write this code to do something like (1) send `planXml` if it's not null to maintain compat with current extensions (2) otherwise since `event.params`.  This would allow you to avoid special casing the visualize event.

Reply:
updated to avoid the 'visualize' special-case.